x/concentrated-liquidity/swaps_tick_cross_test.go

```diff
@@ -755,3 +877,96 @@ func (s *SwapTickCrossTestSuite) TestSwapOutGivenIn_Tick_Initialization_And_Cros
 		}
 	})
 }
+
+// TestSwapOutGivenIn_SwapToAllowedBoundaries tests edge case behavior of swapping
+// to min and max ticks.
+func (s *SwapTickCrossTestSuite) TestSwapOutGivenIn_SwapToAllowedBoundaries() {
+	const shouldStayWithinTheSameBucket = false
+
+	var (
+		tokenZeroDenom       = DefaultCoin0.Denom
+		tokeOneDenom         = DefaultCoin1.Denom
+		smallTokenOneCoinIn  = sdk.NewCoin(tokeOneDenom, sdk.NewInt(100))
+		smallTokenZeroCoinIn = sdk.NewCoin(tokenZeroDenom, sdk.NewInt(100))
+	)
+
+	// tests tick crossing behavior around min tick
+	// It first swaps to the min tick, then tries to swap below it and fails.
+	// At the same time, it validates that the liquidity and current tick are updated correctly.
+	// Additionally, it validates that it is still possible to swap right.
+	s.Run("to min tick", func() {
+		s.SetupTest()
+
+		poolId, _ := s.setupPoolAndPositions(tickSpacingOne, defaultTickSpacingsAway)
+
+		// Fetch pool
+		pool, err := s.App.ConcentratedLiquidityKeeper.GetPoolById(s.Ctx, poolId)
+		s.Require().NoError(err)
+
+		// Compute tokenIn amount necessary to reach the min tick.
+		const isZeroForOne = true
+		tokenIn, _, _ := s.computeSwapAmounts(poolId, sdk.Dec{}, types.MinInitializedTick, isZeroForOne, shouldStayWithinTheSameBucket)
+
+		// Swap the computed large amount.
+		s.swapZeroForOneLeft(poolId, sdk.NewCoin(tokenZeroDenom, tokenIn.Ceil().TruncateInt()))
+
+		// Assert that min tick is crossed and liquidity is zero.
```

Comment:
Just to be clear, it isn't "crossed", it is more so "entered" right? Seems like crossing would be invalid

Reply:
What is the difference between crossing and entering? I'm referring to crossing in the context of our swap logic. The definition in my mental model is compatible with entering

Reply:
Ah I think I see both Matt's confusion as well as mine. We are seeing a tick as a single point, where crossing implies entering and exiting. IMO entering is a better word for this, since even as a bucket, entering implies not exiting, whereas crossing (like "crossing a street") could imply entering and exiting.

Reply:
Like, a street is not a single point, it could be seen as a bucket, and you enter and exit it. Maybe I am just tired but this analogy makes sense to me lol

Reply:
That makes sense but I still think that crossing is correct here. Tick is a single point that maps to the price. However, ticks are crossed to exit one bucket and enter another.

Please see: https://github.com/osmosis-labs/osmosis/blob/892b73d885cb92f8c8857fc2399c259d8b64ad8d/x/concentrated-liquidity/README.md?plain=1#L1698-L1713

Let me know if that makes sense. Happy to add the definition of crossing to terminology and/or discuss changing definitions in a separate PR

Reply:
I am still a bit confused 😭 
The confusing point for me is that if crossing a tick means to exit one bucket and enter another, if we're crossing the min tick, which tick / which bucket are we entering?

Reply:
Let's look at the definition of "Full range": `[MinInitializedTick, MaxTick)`

For min tick, this PR is the solution. Previously, it was impossible to cross `MinInitializedTick`. That's why we introduced the `MinCurrentTick` which is one smaller than the `MinInitializedTick`.

Only `MinInitializedTick` is crossed, you end up on `MinCurrentTick` that has no liquidity.

With `MaxTick`, since the definition of full range is exclusive of the upper bound, you end up on the MaxTick itself once you cross it.

However, it is fair to say that once these edges are crossed, we are in no bucket. We completely left the allowed range for buckets and have no liquidity. We are located either on the `MinCurrentTick` or `MaxTick`. 

From that point, you can only swap left to enter the first bucket for `MaxTick` or swap right to enter the first bucket for `MinTick`

Reply:
I created a PR to document everything we discussed here: https://github.com/osmosis-labs/osmosis/pull/5595